pkg/api/aim/experiments.go

```diff
@@ -327,3 +335,77 @@ func DeleteExperiment(c *fiber.Ctx) error {
 		"status": "OK",
 	})
 }
+
+func UpdateExperiment(c *fiber.Ctx) error {
+	ns, err := namespace.GetNamespaceFromContext(c.Context())
+	if err != nil {
+		return api.NewInternalError("error getting namespace from context")
+	}
+	log.Debugf("updateExperiment namespace: %s", ns.Code)
+
+	params := struct {
+		ID string `params:"id"`
+	}{}
+	if err = c.ParamsParser(&params); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+	id, err := strconv.ParseInt(params.ID, 10, 32)
+	if err != nil {
+		return api.NewBadRequestError("Unable to parse experiment id '%s': %s", params.ID, err)
+	}
+
+	var updateRequest request.UpdateExperimentRequest
+	if err = c.BodyParser(&updateRequest); err != nil {
+		return fiber.NewError(fiber.StatusUnprocessableEntity, err.Error())
+	}
+
+	experimentRepository := repositories.NewExperimentRepository(database.DB)
+	tagRepository := repositories.NewTagRepository(database.DB)
+	experiment, err := experimentRepository.GetByNamespaceIDAndExperimentID(c.Context(), ns.ID, int32(id))
+	if err != nil {
+		return fiber.NewError(
+			fiber.StatusInternalServerError, fmt.Sprintf("unable to find experiment '%s': %s", params.ID, err),
+		)
+	}
+	if experiment == nil {
+		return fiber.NewError(fiber.StatusNotFound, fmt.Sprintf("unable to find experiment '%s'", params.ID))
+	}
+	if updateRequest.Archived != nil {
+		if *updateRequest.Archived {
+			experiment.LifecycleStage = models.LifecycleStageDeleted
+		} else {
+			experiment.LifecycleStage = models.LifecycleStageActive
+		}
+	}
+
+	if updateRequest.Name != nil {
+		experiment.Name = *updateRequest.Name
+	}
+
+	if updateRequest.Archived != nil || updateRequest.Name != nil {
+		if err := database.DB.Transaction(func(tx *gorm.DB) error {
+			if err := experimentRepository.UpdateWithTransaction(c.Context(), tx, experiment); err != nil {
+				return err
+			}
+			return nil
+		}); err != nil {
+			return fiber.NewError(fiber.StatusInternalServerError,
+				fmt.Sprintf("unable to update experiment %q: %s", params.ID, err))
+		}
+	}
+	if updateRequest.Description != nil {
+		description := models.ExperimentTag{
+			Key:          common.DescriptionTagKey,
+			Value:        *updateRequest.Description,
+			ExperimentID: *experiment.ID,
+		}
+		if err := tagRepository.CreateExperimentTag(c.Context(), &description); err != nil {
```

Comment:
for update, wouldn't we want to update the description tag if it exists instead of creating a new one?

Reply:
this should already update the existing tag if it already exists:
https://github.com/G-Research/fasttrackml/blob/202fa759a4704606bb21f7f6e7558e7920f583a1/pkg/api/mlflow/dao/repositories/tag.go#L43